### ISSUE TITLE
How do we handle resourceGroupName / resource_group definition?

### ISSUE THREAD

**Issue by zikalino** [Azure/magic-module-specs#34]
It seems like there's some kind of exception built in and it's not mentioned in current documentation.

In batch account example **api.yaml** has **resourceGroupName** but in generated module we have **resource_group**.

I think the mechanism should be included in **api.yaml** as an exception.

Also it's a bit confusing in examples: **resourceGroup** and **resource_group_name**, but I described this in another issue related to examples as well.

**Comment by JunyiYi** [comment]
Yes, `resourceGroupName` is a very special string which will be handled specifically in MM for both Terraform and Ansible. But since you are generating `resourceGroup`, we will use overrides in terraform.yaml to rename it to `resourceGroupName`.
